src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -98,7 +104,11 @@ protected override Task DisposeCoreAsync(bool initialized)
             // active configuration should be updated before resetting subscriptions
             await RefreshActiveConfigurationAsync().ConfigureAwait(false);
 
-            _designTimeBuildSubscriptionLink?.Dispose();
+            foreach (var disposable in _designTimeBuildSubscriptionLinks)
+            {
+                disposable.Dispose();
+            }
+            _designTimeBuildSubscriptionLinks.Clear();
```

Comment:
Hmm ya but it's not just this link, this entire ResetSubscriptions method shouldn't be doing it's work if the object is disposed. Is there anything in the OnceInitialized baseclass that I can use to ensure that or do I need a giant lock around this method? @jviau?